### DISCUSSION TITLE
Extending .vercelignore from other files

### DISCUSSION THREAD

**Discussion by balthild** [vercel/vercel#4679] [Ideas]
Currently I have to write ignored entries twice in `.vercelignore` and `.gitignore`, even if they're identical.

I suggest that vercel cli could support extending ignore files. For example, if I have the `.gitignore`:

```
/dist
```

and the `.vercelignore`

```
#include ".gitignore"

/.env.example
```

then vercel cli should ignore both `/dist` and `/.env.example`.

**Comment by styfle** [comment]
Hi @balthild 

If you connect your git repository using the [Vercel Git Integration](https://vercel.com/docs/v2/git-integrations), then you can deploy with `git push` and the `.gitignore` will be respected.

In this scenario, your `.vercelignore` will effectively be an extension of `.gitignore` because the integration only clones the files that were committed to your git repository.

**Comment by balthild** [comment]
Yes, but sometimes I would create testing deployments from local during the development, for example, showing the intermediate revision of the project to others.

Also, some of my projects are hosted at a standalone git server, which have not supported by vercel yet. (Though we can set up a CI server that deploys to vercel.)

**Comment by styfle** [comment]
> for example, showing the intermediate revision of the project to others.

This can be achieved by creating a git branch for your test which will create a Preview Deployment when you `git push`. Preview deployments have unique URLs that you can share with others for that particular commit. If you create a PR, it will have a unique URL for the latest push in the branch so you can share that Preview URL once for the entire length of that feature's development.

https://vercel.com/docs/v2/platform/deployments#deployment-types

> some of my projects are hosted at a standalone git server, which have not supported by vercel yet

Which standalone git server are you using?

**Comment by balthild** [comment]
I'm using a Gitea set up on our team's private server. When I try importing project from the git url, it says `Error:Invalid repository URL.`

**Comment by styfle** [comment]
Is the git url public or does it require auth to clone?

**Comment by balthild** [comment]
It requires auth to clone.

**Comment by styfle** [comment]
I see. So we will need to add a new feature to input git credentials for cloning, unless Gitea has some sort of OAuth implementation like GitHub/GitLab.
